pyomo/contrib/mindtpy/cut_generation.py

```diff
@@ -216,66 +140,67 @@ def add_ecp_cuts(target_model, solve_data, config,
-                    or (config.linearize_inactive and lower_slack > 0):
-                if config.add_slack:
-                    slack_var = target_model.MindtPy_utils.MindtPy_linear_cuts.slack_vars.add()
-
-                target_model.MindtPy_utils.MindtPy_linear_cuts.ecp_cuts.add(
-                    expr=(sum(value(jacs[constr][var])*(var - var.value)
-                              for var in constr_vars)
-                          + (slack_var if config.add_slack else 0)
-                          >= -lower_slack)
-                )
+            if constr.has_ub():
+                try:
+                    upper_slack = constr.uslack()
+                except (ValueError, OverflowError):
+                    config.logger.warning(
+                        'constraint {} has caused either a '
+                        'ValueError or OverflowError.'
+                        '\n'.format(
+                            constr))
+                    continue
+                if (linearize_active and abs(upper_slack) < config.ecp_tolerance) \
+                        or (linearize_violated and upper_slack < 0) \
+                        or (config.linearize_inactive and upper_slack > 0):
+                    if config.add_slack:
+                        slack_var = target_model.MindtPy_utils.MindtPy_linear_cuts.slack_vars.add()
+
+                    target_model.MindtPy_utils.MindtPy_linear_cuts.ecp_cuts.add(
+                        expr=(sum(value(jacs[constr][var])*(var - var.value)
+                                  for var in constr_vars)
+                              - (slack_var if config.add_slack else 0)
+                              <= upper_slack)
+                    )
+
+            if constr.has_lb():
+                try:
+                    lower_slack = constr.lslack()
+                except (ValueError, OverflowError):
+                    config.logger.warning(
+                        'constraint {} has caused either a '
+                        'ValueError or OverflowError.'
+                        '\n'.format(
+                            constr))
+                    continue
+                if (linearize_active and abs(lower_slack) < config.ecp_tolerance) \
+                        or (linearize_violated and lower_slack < 0) \
+                        or (config.linearize_inactive and lower_slack > 0):
+                    if config.add_slack:
+                        slack_var = target_model.MindtPy_utils.MindtPy_linear_cuts.slack_vars.add()
+
+                    target_model.MindtPy_utils.MindtPy_linear_cuts.ecp_cuts.add(
+                        expr=(sum(value(jacs[constr][var])*(var - var.value)
+                                  for var in constr_vars)
+                              + (slack_var if config.add_slack else 0)
+                              >= -lower_slack)
+                    )
 
 # def add_oa_equality_relaxation(var_values, duals, solve_data, config, ignore_integrality=False):
```

Comment:
This is still commented? We should consider if it is useful for anything and if not, delete it.